storage/range_test.go

```diff
@@ -2733,12 +2733,13 @@ func (mrm *mockRangeManager) ProposeRaftCommand(idKey cmdIDKey, cmd proto.Intern
 	return mrm.mockProposeRaftCommand(idKey, cmd)
 }
 
-// TestRaftGroupDeletedError verifies that a request leader proposal which fails with
+// TestRequestLeaderEncounterGroupDeleteError verifies that a request leader proposal which fails with
 // multiraft.ErrGroupDeleted is converted to a RangeNotFoundError in the Store.
 func TestRequestLeaderEncounterGroupDeleteError(t *testing.T) {
 	defer leaktest.AfterTest(t)
-
-	tc := testContext{}
+	tc := testContext{
+		rng: &Range{},
```

Comment:
It suppresses bootstrapping of the first range, which is what allows this test to create a range later instead of modifying a range created by the store (which we apparently don't have a non-racy way to do). It might be cleaner to add a third value `bootstrapNone` to the `bootstrapMode` enum.